### PR TITLE
NMS-12513 Open Redirect security issue

### PR DESCRIPTION
**Pull request by patrick-schweizer**
This Pull Request removes open redirect vulnerabilities by restricting the redirect to a whitelist.
To be tested: I hope I found all links for the whitelist but I might have missed some.

### External References

* JIRA (Issue Tracker): https://issues.opennms.org/browse/NMS-12612
* Bamboo (Continuous Integration): https://bamboo.opennms.org/

### CODE REVIEW

**Comment by j-white** [comment]
Given there's some risk in breaking some existing workflows with this, I would re-target this to the `release-26.0.0` branch. We can then back-port for `foundation-2019` at a later time.

File: opennms-webapp/src/main/java/org/opennms/web/controller/alarm/AcknowledgeAlarmByFilterController.java [hunk 1]
```diff
@@ -61,7 +62,10 @@
     private AlarmRepository m_webAlarmRepository;
     
     private String m_redirectView;
-    
+
+    private RedirectRestricter redirectRestricter = RedirectRestricter.builder()
```
Comment (j-white) [inline opennms-webapp/src/main/java/org/opennms/web/controller/alarm/AcknowledgeAlarmByFilterController.java]: No allowed redirects? Is this intentional?

Comment (patrick-schweizer) [reply opennms-webapp/src/main/java/org/opennms/web/controller/alarm/AcknowledgeAlarmByFilterController.java]: yes, I didn't find any link with a redirect pointing to this controller (although I might have missed one). The alternative would be to remove the redirect parameter completely from this controller (would result in the same thing but viewer code). I wonder why we have the redirect parameter if we don't use it:
Either this is just a copy and paste leftover or
I didn't find the link...

Comment (patrick-schweizer) [reply opennms-webapp/src/main/java/org/opennms/web/controller/alarm/AcknowledgeAlarmByFilterController.java]: TODO: Patrick add as comment in code

Comment (patrick-schweizer) [reply opennms-webapp/src/main/java/org/opennms/web/controller/alarm/AcknowledgeAlarmByFilterController.java]: done.

**Pull request merged**